### ISSUE TITLE
Allow updating dynamic fields (e.g. based on changed fields name)

### ISSUE THREAD

**Issue by swernerx** [final-form/final-form-calculate#5]
### Are you submitting a **bug report** or a **feature request**?

Feature Request

### What is the current behavior?

Right now when combined with final-form-arrays there is no possibility of changing fields in the same index position as the one which is matched. Based on the online example I tried the following:

```js
const calculator = createDecorator(
    {
      field: /timeFrom/, // when minimum changes...
      updates: {
        // ...update maximum to the result of this function
        maximum: (timeFrom, allValues) => {
            return Math.max(timeFrom || 0, allValues.timeTo || 0);
        }
      }
    }
);
```

What I need instead of `allValues.timeTo` would be something like `allValues.someField[2].timeTo`. All I need for this I need the fieldName of the field matched by the decorator. 

What about passing this to the methods in "updates"? This would definitely change the semantics/signature. I would suggest that this should work:

```js
const calculator = createDecorator(
    {
      field: /timeFrom/, // when minimum changes...
      updates: (fieldValue, fieldName, allValues) => {
        const targetField = fieldName.replace("timeFrom", "timeTo");
        const toFieldValue = get(targetField, allValues);
        set(targetField, allValues, Math.max(fieldValue, toFieldValue));
      }
    }
);
```

So additionally to having a static map of updates, there is also the possibility to update fields based on the incoming field name.

### What's your environment?

grep final  package.json 
    "final-form": "^4.2.1",
    "final-form-arrays": "^1.0.4",
    "final-form-calculate": "^1.0.2",
    "react-final-form": "^3.1.2",
    "react-final-form-arrays": "^1.0.4",

npm --version
5.7.1

node --version
v9.6.1

**Comment by swernerx** [comment]
BTW: I also wonder whether there is some generic set/get. Right now I just use the methods from lodash to read from deeper nested structures.

**Comment by erikras** [comment]
Published fix in [`v1.1.0`](https://github.com/final-form/final-form-calculate/releases/tag/v1.1.0).
